mm2src/coins/coin_errors.rs

```diff
@@ -0,0 +1,442 @@
+    #[display(fmt = "Invalid address: {}", _0)]
+    InvalidAddress(String),
+    Internal(String),
+    UnsupportedAddr(String),
+    MethodNotSupported(String),
+    ScriptHashTypeNotSupported {
+        script_hash_type: String,
+    },
+    CashAddressErr(String),
+    #[display(fmt = "Transaction Reading Error: {}", _0)]
+    TxReadError(String),
+    UnexpectedDerivationMethod(UnexpectedDerivationMethod),
+    UtxoRpcError(String),
+}
+
+impl From<AddressParseError> for MyAddressError {
+    fn from(e: AddressParseError) -> Self { MyAddressError::AddressParseError(e) }
+}
+
+impl From<UnexpectedDerivationMethod> for MyAddressError {
+    fn from(e: UnexpectedDerivationMethod) -> Self { MyAddressError::UnexpectedDerivationMethod(e) }
+}
+
+impl From<keys::Error> for MyAddressError {
+    fn from(e: keys::Error) -> Self { MyAddressError::Internal(e.to_string()) }
+}
+
+impl From<UnsupportedAddr> for MyAddressError {
+    fn from(e: UnsupportedAddr) -> Self { MyAddressError::UnsupportedAddr(e.to_string()) }
+}
+
+#[derive(Debug, Display, PartialEq)]
+pub enum AddressParseError {
+    AddrFormatParseError(String),
+    AddrDisplayError(String),
+    AddrConversionErr(String),
+    CashAddressErr(String),
+    EncodeError(String),
+    InvalidHexError(String),
+    Internal(String),
+    #[display(fmt = "Invalid address: {}", _0)]
+    InvalidAddress(String),
+    #[display(fmt = "Invalid address checksum")]
+    InvalidAddressCheckSum,
+    InvalidAddressPrefix(String),
+    MyAddressError(String),
+    PlatformConfIsNull(String),
+    ParsingError(String),
+    UnsupportedAddr(String),
+    UnsupportedProtocol(String),
+    UnexpectedProtocol(String),
+    ScriptHashTypeNotSupported {
+        script_hash_type: String,
+    },
+    #[display(fmt = "Transaction Reading Error: {}", _0)]
+    UnexpectedDerivationMethod(UnexpectedDerivationMethod),
+    #[display(fmt = "Topic {:?} is expected to be H256 encoded topic (with length of 64)", _0)]
+    UnexpectedTopicLen(String),
+    UtxoConfError(String),
+}
+
+impl From<keys::Error> for AddressParseError {
+    fn from(e: keys::Error) -> Self { AddressParseError::Internal(e.to_string()) }
+}
+
+impl From<MyAddressError> for AddressParseError {
+    fn from(e: MyAddressError) -> Self { AddressParseError::MyAddressError(e.to_string()) }
+}
+
+impl From<FromHexError> for AddressParseError {
+    fn from(e: FromHexError) -> Self { AddressParseError::InvalidHexError(e.to_string()) }
+}
+
+impl From<UtxoConfError> for AddressParseError {
+    fn from(e: UtxoConfError) -> Self { AddressParseError::UtxoConfError(e.to_string()) }
+}
+
+impl From<serde_json::Error> for AddressParseError {
+    fn from(e: serde_json::Error) -> Self { AddressParseError::Internal(e.to_string()) }
+}
+
+impl From<secp256k1::Error> for AddressParseError {
+    fn from(e: secp256k1::Error) -> Self { AddressParseError::Internal(e.to_string()) }
+}
+
+#[derive(Display, Debug)]
+pub enum SignedEthTxError {
```

Comment:
This error seems to be related to `ETH` coin only. Someday I dream we move `eth.rs` to a separate crate.
Please move it to eth.rs or `eth_errors.rs`